### ISSUE TITLE
Add documentations

### ISSUE THREAD

**Issue by ayrat555** [mana-ethereum/ethereumex#11]
The library is purely documented. Fix that

**Comment by hickscorp** [comment]
It'd be nice to add a small paragraph in the docs on how to use the lib to perform smart contracts method calls, given a JSON ABI.

**Comment by hayesgm** [comment]
Sure, I'll work on adding this to the README, but for those curious, it'll be something like this:

Add https://hex.pm/packages/blockchain and https://hex.pm/packages/abi

```elixir
abi_encoded_data = ABI.encode("baz(uint,address)", [50, <<1::160>> |> :binary.decode_unsigned])

transaction_data = %Blockchain.Transaction{
    data: abi_encoded_data,
    gas_limit: 100_000,
    gas_price: 16_000_000_000,
    init: <<>>,
    nonce: 5,
    to: <<>>,
    value: 0
}
|> Blockchain.Transaction.Signature.sign_transaction(private_key)
|> Blockchain.Transaction.serialize()
|> ExRLP.encode()
|> Base.encode16(case: :lower)

Ethereumex.HttpClient.eth_send_raw_transaction("0x" <> transaction_data)
```

**Comment by hickscorp** [comment]
That would be most helpful indeed - and IMO small things like this are the only roadblock preventing more exposure of this great library.

**Comment by tzumby** [comment]
@hayesgm did you get a chance to add that snipped to the README ? I went through a similar process to get the eth_call working. I can submit a PR with the docs and include your snippet as well.

**Comment by ayrat555** [comment]
@tzumby it'd be great

**Comment by tzumby** [comment]
@ayrat555 PR is in :)

**Comment by ghbutton** [comment]
Can we close this issue if it has been resolved?

**Comment by ayrat555** [comment]
@ghbutton After adding doctests and moduledocs we can close it

**Comment by hayesgm** [comment]
Would doctests be difficult since they are pretty side-effecting? Doctests are great for pure functions, but they aren't as clean for impure code.

**Comment by hickscorp** [comment]
@hayesgm I usually go for module doc for such examples. Maybe that's the way to go.

**Comment by ayrat555** [comment]
Good point, Geoff. I think just descriptions for methods and their params would be nice
